agents/pfeiffer_turbo_controller/Pfeiffer_Turbo_Controller_Agent.py

```diff
@@ -0,0 +1,206 @@
+                #self.idn = self.psu.identify()
+            except socket.timeout as e:
+                self.log.error("Turbo Controller timed out during connect")
+                return False, "Timeout"
+            self.log.info("Connected to turbo controller")
+            
+                                  
+        # Start data acquisition if requested in site-config
+        auto_acquire = params.get('auto_acquire', False)
+        print(f"auto acquire is {auto_acquire}")
+        if auto_acquire:
+            self.agent.start('acq')
+
+        return True, 'Initialized Turbo Controller.'
+
+    def monitor_turbo(self, session, params=None):
+        """
+            Process to continuously monitor turbo motor temp and rotation speed and
+            send info to aggregator.
+
+            Args:
+                wait (float, optional):
+                    time to wait between measurements [seconds].
+        """
+        if params is None:
+            params = {}
+
+        wait_time = params.get('wait', 1)
+
+        self.monitor = True
+
+        while self.monitor:
+            with self.lock.acquire_timeout(1) as acquired:
+                if acquired:
+                    data = {
+                        'timestamp': time.time(),
+                        'block_name': 'turbo_output',
+                        'data': {}
+                    }
+                    
+                    try:    
+                        data['data']["Turbo_Motor_Temp"] = self.turbo.get_turbo_motor_temperature()
+                        data['data']["Rotation_Speed"] = self.turbo.get_turbo_actual_rotation_speed()
+                    
+                    except ValueError as e:
+                        self.log.error(e)
+                        
+                    # self.log.info(str(data))
+                    # print(data)
+                    self.agent.publish_to_feed('pfeiffer_turbo', data)
+
+                    # Allow this process to be queried to return current data
+                    session.data = data
+
+                else:
+                    self.log.warn("Could not acquire in monitor turbo")
+
+            time.sleep(wait_time)
+
+        return True, "Finished monitoring turbo"
+
+    def stop_monitoring(self, session, params=None):
+        """Stop monitoring the turbo output."""
+        if self.monitor:
+            self.monitor = False
+            return True, 'requested to stop taking data.'
+        else:
+            return False, 'acq is not currently running'
+
+    def turn_turbo_on(self, session, params=None):
+        """Turns the turbo on."""
+
+        with self.lock.acquire_timeout(1) as acquired:
+            if acquired:
+                self.turbo.ready_turbo()
+                time.sleep(1)
+                self.turbo.turn_turbo_motor_on()
+            else:
+                return False, "Could not acquire lock"
+
+        return True, 'Turned Turbo Motor On.'
```

Comment:
I make some suggestions in the driver file near these methods. But it looks to me like if these fail for whatever reason, maybe a current error on the turbo preventing it from being readied and turned on, this would still report that the task succeeded in turning on the turbo motor. At minimum I think we should be checking the returned values from these calls to control the turbo.

Reply:
I put in two checks, one for 'ready', one for 'on'